packages/trader/src/Stores/Modules/Trading/trade-store.js

```diff
@@ -271,7 +271,9 @@ export default class TradeStore extends BaseStore {
 
         await this.setActiveSymbols();
         await this.root_store.active_symbols.setActiveSymbols();
-        if (should_set_default_symbol) await this.setDefaultSymbol();
+        if (should_set_default_symbol && !this.root_store.common.is_language_changing) {
+            await this.setDefaultSymbol();
```

Comment:
I think it not so clean function. But anyway, thank for your remark. I agree with your thoughts .
 I moved is_language_changing in outer function.

Reply:
@yauheni-kryzhyk-deriv Thank you) yes, sure, this is not a completely pure function because it's async and it awaits API response after it sets a symbol but in terms of loading a default or non-default symbol it has to depend exclusively on incoming params as it did before.
Btw since you don't change this function anymore, you can leave it as it was (without curly brackets). 🙂